### PR TITLE
More verdeps testing changes

### PR DESCRIPTION
**Pull request by skeswa**
- Added a standard github request service mock
- Fixed the `helpers_test.go` linter errors (mostly by moving and redesigning `MockFileInfo`)
- Re-wrote `bytes_diffs_test.go` to use `goconvey`
- Re-formatted `fetch_sha.go` to stay within 80 characters per line (within reason)
- Wrote tests for `fetch_sha.go`

### CODE REVIEW

File: router/download_package_test.go [hunk 1]
```diff
@@ -326,11 +326,11 @@ func TestDownloadPackage(t *testing.T) {
...
 		Return([]os.FileInfo{
-			io.NewMockFileInfo("archive.zip", 1234),
-			io.NewMockFileInfo("archive.zip", 1235),
-			io.NewMockFileInfo("archive.zip", 1236),
-			io.NewMockFileInfo("akdjshfgaldfkjhjdfhgaksjhfg", 9999),
-			io.NewMockFileInfo("archive.zip", 1237),
+			io.NewFakeFileInfo("archive.zip", 1234, false),
+			io.NewFakeFileInfo("archive.zip", 1235, false),
+			io.NewFakeFileInfo("archive.zip", 1236, false),
+			io.NewFakeFileInfo("akdjshfgaldfkjhjdfhgaksjhfg", 9999, false),
```
Comment (Shikkic) [inline router/download_package_test.go]: Thought you hated buldge?

File: lib/io/fake.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package io
```
Comment (ZacharyThomas) [inline lib/io/fake.go]: >fake.go 

real talk we should rename these files

File: router/download_package_test.go (hunk 1 above)
Comment (skeswa) [inline router/download_package_test.go]: it was important they are out of order.

**Comment by ZacharyThomas** [comment]
I don't like BDD because I am scum but okay. lgtm

**Pull request merged**